src/components/molecules/DataWrapper.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+import { ApolloError, DocumentNode, useQuery } from '@apollo/client';
+import { TimeEntry } from '../atoms/ThresholdChart';
+import { formatMetData } from '../../utils/formatMetData';
+
+export type DataWrapperProps = {
+    query: DocumentNode;
+    children: (args: { data: TimeEntry[] | undefined; loading: boolean; error: ApolloError | undefined }) => ReactNode;
+};
+
+const DataWrapper: React.FC<DataWrapperProps> = ({ children, query }) => {
+    const [formattedData, setFormattedData] = useState<TimeEntry[] | undefined>(undefined);
+    const { loading, error, data } = useQuery(query);
+
+    useEffect(() => {
+        if (data) setFormattedData(formatMetData(data));
```

Comment:
Here you limit this component to only working with `metData`. Could you pass this mapping function in from props?

Reply:
That could work. Were thinking about using a switch based on the query to know which format that should be used. Not sure what the best approach is.

Reply:
Generics! They are a perfect fit for this (at least I think so):)